app-backend/api/src/main/scala/uploads/Routes.scala

```diff
@@ -172,17 +173,24 @@ trait UploadRoutes
     authorizeAsync {
       UploadDao.query.ownedBy(user, uploadId).exists.transact(xa).unsafeToFuture
     } {
-      extractTokenHeader { jwt =>
-        onSuccess(
-          UploadDao.query
-            .filter(uploadId)
-            .selectOption
-            .transact(xa)
-            .unsafeToFuture) {
-          case Some(_) =>
-            complete(CredentialsService.getCredentials(user, uploadId, jwt))
-          case None => complete(StatusCodes.NotFound)
-        }
+      extractTokenHeader {
+        case Some(jwt) =>
+          onSuccess(
+            UploadDao.query
+              .filter(uploadId)
+              .selectOption
+              .transact(xa)
+              .unsafeToFuture) {
+            case Some(_) =>
+              complete(CredentialsService.getCredentials(user, uploadId, jwt))
+            case None => complete(StatusCodes.NotFound)
+          }
+        case _ =>
```

Comment:
This takes the `reject` branch of what `extractTokenHeader` used to have and moves it into this Route, since this is the only place that `extractTokenHeader` was used in routing previously. I needed `extractTokenHeader` not to reject if the token was missing to be able to do authorization based on tokens or map tokens or public